api/src/test/java/io/minio/MinioClientTest.java

```diff
@@ -253,6 +270,273 @@ public void testGetObject()
   }
 
   @Test
+  public void testGetObjectEncryptedAes()
+      throws NoSuchAlgorithmException, InvalidKeyException, IOException, XmlPullParserException, MinioException,
+      NoSuchPaddingException, IllegalBlockSizeException, BadPaddingException, InvalidAlgorithmParameterException {
+    MockWebServer server = new MockWebServer();
+    MockResponse response1 = new MockResponse();
+    MockResponse response2 = new MockResponse();
+    final String expectedObject = HELLO_WORLD;
+
+    // Generate Master key
+    KeyGenerator symKeyGenerator = KeyGenerator.getInstance("AES");
+    symKeyGenerator.init(256);
+    SecretKey symMasterKey = symKeyGenerator.generateKey();
+    EncryptionMaterials encMaterials = new EncryptionMaterials(symMasterKey);
+
+    // Generate object encryption key
+    SecretKey symDataKey = symKeyGenerator.generateKey();
+    byte[] plainDataKey = symDataKey.getEncoded();
+
+    // Generate an iv to be used for encryption
+    SecureRandom ivSeed = new SecureRandom();
+    byte[] iv = new byte[16];
+    ivSeed.nextBytes(iv);
+
+    // Create data key encryption cipher
+    Cipher keyEncryptionCipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
+
+    // init cipher with mode and master key
+    keyEncryptionCipher.init(Cipher.ENCRYPT_MODE, symMasterKey);
+    // Encrypt the data key
+    byte[] encryptedDataKey = keyEncryptionCipher.doFinal(plainDataKey);
+
+    String encDataKey = new String(Base64.encodeBase64(encryptedDataKey), StandardCharsets.UTF_8);
+    String ivString = new String(Base64.encodeBase64(iv), StandardCharsets.UTF_8);
```

Comment:
`Digest.java` has some methods related to base64 hashing? Is it possible to use/enhance them?